src/backend/utils/init/postinit.c

```diff
@@ -368,6 +368,22 @@ InitPostgres(const char *in_dbname, Oid dboid, const char *username,
 	char		dbname[NAMEDATALEN];
 
 	/*
+	 * User is not supposed to connect to hcatalog database,
+	 * because it's reserved for hcatalog feature integration
+	 */
+	if (!bootstrap)
+	{
+		if (strcmp(in_dbname, HcatalogDbName) == 0)
+		{
+			ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_DATABASE),
```

Comment:
ERRCODE_RESERVED_NAME error code is being referenced when trying to create/update object, so ERRCODE_UNDEFINED_DATABASE should be fine for this case. Also in current implementation, when users tries to  connect "hcatalog" database he is getting ERRCODE_UNDEFINED_DATABASE.